airflow/providers/sqlite/hooks/sqlite.py

```diff
@@ -33,13 +34,16 @@ class SqliteHook(DbApiHook):
 
     def get_conn(self) -> sqlite3.dbapi2.Connection:
         """Returns a sqlite connection object"""
-        conn_id = getattr(self, self.conn_name_attr)
-        airflow_conn = self.get_connection(conn_id)
-        conn = sqlite3.connect(airflow_conn.host)
+        # The sqlite3 connection does not use the sqlite prefix.
+        # See https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#connect-strings for details.
+        uri = self.get_uri().replace("sqlite:///", "")
+        conn = sqlite3.connect(uri)
         return conn
 
     def get_uri(self) -> str:
         """Override DbApiHook get_uri method for get_sqlalchemy_engine()"""
-        conn_id = getattr(self, self.conn_name_attr)
-        airflow_conn = self.get_connection(conn_id)
-        return f"sqlite:///{airflow_conn.host}"
+        # The sqlite connection has one more slash for path specification.
+        # See https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#uri-connections for details.
+        uri = super().get_uri().replace("sqlite://", "sqlite:///")
+        raw_uri = unquote(uri)
+        return raw_uri
```

Comment:
i guess since it is a provider, we can break things, but you'll have to make it major release... i think the way to signal that to release manager is just add to changelog ?? @eladkal

Reply:
If we can't deprecate first and must do a major release then we need to update provider.yaml major version and add relevant entry in the provider change log as part of this PR.

Reply:
See line 139-140.

Reply:
But what changes is what I mentioned in https://github.com/apache/airflow/pull/28721#issuecomment-1370717908

Reply:
breaking change can also be about changing the input schema.
I did not review the PR, I just commented that if this is a breaking change then just need adding entry in change log and updating provider.yaml if this is not a breaking change but just a bug fix then no further action is required.